include/LinearAlgebra/sparseLUSolver.h

```diff
@@ -237,7 +252,7 @@ void SparseLUSolver<T>::solveInPlace(T* b) const
                 b[i] -= U_values[idx] * b[col];
             }
         }
-        if (std::abs(diag) < 1e-12) {
+        if (std::abs(diag) < 1e-15) {
             std::cerr << "Zero diagonal encountered in U at row " << i << "!\n";
             std::exit(EXIT_FAILURE);
         }
```

Comment:
I am curious about the decision to use an if block in the `solveInPlace` method. Normally `factorize` is called once per matrix, while `solveInPlace` may be called multiple times. It therefore seems like it would make more sense to place this block in `factorize`, unless you expect users to occasionally modify the values of `U_vlaues` themselves?

Reply:
I also wondered about assertion but I think an explicit error makes sense as the contents of the matrix may depend on user input. If this is the case then the error should appear even in release mode